src/hw-cbmc/hw_cbmc_parse_options.cpp

```diff
@@ -50,66 +59,80 @@ int hw_cbmc_parse_optionst::doit()
 
   optionst options;
   get_command_line_options(options);
-  eval_verbosity();
+
+  messaget::eval_verbosity(cmdline.get_value("verbosity"),
+                           messaget::M_STATISTICS, ui_message_handler);
 
   //
   // Print a banner
   //
-  status() << "HW-CBMC version " CBMC_VERSION << eom;
+  log.status() << "HW-CBMC version " << CBMC_VERSION << messaget::eom;
 
   register_languages();
 
   if(cmdline.isset("preprocess"))
   {
-    preprocessing();
+    preprocessing(options);
     return 0;
   }
 
   if(cmdline.isset("vcd"))
     options.set_option("vcd", cmdline.get_value("vcd"));
 
-  cbmc_solverst cbmc_solvers(options, symbol_table, ui_message_handler);
-  cbmc_solvers.set_ui(get_ui());
-  std::unique_ptr<cbmc_solverst::solvert> cbmc_solver;
-  
-  try
-  {
-    cbmc_solver=cbmc_solvers.get_solver();
-  }
-  
-  catch(const char *error_msg)
-  {
-    error() << error_msg << eom;
-    return 1;
-  }
-
-  prop_convt &prop_conv=cbmc_solver->prop_conv();
-  hw_bmct hw_bmc(options, symbol_table, ui_message_handler, prop_conv);
+  std::unique_ptr<goto_verifiert> verifier = nullptr;
+  verifier = util_make_unique<
+      all_properties_verifier_with_trace_storaget<multi_path_symex_checkert>>(
+      options, ui_message_handler, goto_model);
 
-  goto_functionst goto_functions;
+  solver_factoryt sf{options, namespacet{goto_model.symbol_table},
+                     ui_message_handler, false};
+  auto my_solver = sf.get_solver();
+  prop_convt &prop_conv =
+      static_cast<prop_convt &>(*my_solver->decision_procedure_ptr);
 
-  int get_goto_program_ret=get_goto_program(
-    options, hw_bmc.bmc_constraints, goto_functions);
-  if(get_goto_program_ret!=-1)
+  int get_goto_program_ret =
+      get_goto_program(goto_model, options, cmdline, ui_message_handler);
+  if (get_goto_program_ret != -1)
     return get_goto_program_ret;
 
-  hw_bmc.unwind_no_timeframes=get_bound();
-  hw_bmc.unwind_module=get_top_module();
+  std::list<exprt> constraints;
+  int get_modules_ret = get_modules(constraints);
+  if (get_modules_ret != -1)
+    return get_modules_ret;
 
-  label_properties(goto_functions);
+  goto_convert(goto_model.symbol_table, goto_model.goto_functions,
+               ui_message_handler);
+  if (cbmc_parse_optionst::process_goto_program(goto_model, options, log))
+    return CPROVER_EXIT_INTERNAL_ERROR;
 
-  if(cmdline.isset("show-properties"))
-  {
-    const namespacet ns(symbol_table);
-    show_properties(ns, get_ui(), goto_functions);
-    return 0;
+  unwind_no_timeframes = get_bound();
+  unwind_module = get_top_module();
+  do_unwind_module(prop_conv);
+
+  // the 'extra constraints'
+  if (!constraints.empty()) {
+    log.status() << "converting constraints" << messaget::eom;
+
+    for (const auto &constraint : constraints) {
+      prop_conv.set_to_true(constraint);
```

Comment:
I don't think this is doing anything because this is a different solver than used by the verifier. You'd need to write an extension of the goto verifier and checker that allows you to pass in extra constraints.

Reply:
I commented out the extraction of `prop_conv` and added some comments as to what needs to be done next to get this working.